contracts/LiquidityConversionRates.sol

```diff
@@ -3,55 +3,136 @@ pragma solidity ^0.4.18;
 import "./ConversionRates.sol";
 import "./LiquidityFormula.sol";
 
-contract LiquidityConversionRates is ConversionRates, LiquidityFormula {
+contract LiquidityConversionRates is ConversionRatesInterface, LiquidityFormula, Withdrawable, Utils {
+    ERC20 token;
     uint public r;
     uint public Pmin;
-    uint public precision;
+    uint public qtyPrecision;
     uint public numPrecisionBits;
+    uint public maxEthCapBuy;
+    uint public maxEthCapSell;
+    uint public collectedTokenFees;
+    uint public feeInBps;
 
+    function LiquidityConversionRates(address _admin) public{
+        admin = _admin;
```

Comment:
`transferAdminQuickly(_admin)` is more elegant.

Reply:
done